workflow-runtime/build.gradle.kts

```diff
@@ -18,35 +14,51 @@ java {
 
 apply(from = rootProject.file(".buildscript/configure-maven-publish.gradle"))
 
-// Benchmark configuration.
-configure<JMHPluginExtension> {
-  include = listOf(".*")
-  duplicateClassesStrategy = DuplicatesStrategy.WARN
-}
-configurations.named("jmh") {
-  attributes.attribute(Usage.USAGE_ATTRIBUTE, objects.named(Usage.JAVA_RUNTIME))
-}
-tasks.named<KotlinCompile>("compileJmhKotlin") {
-  kotlinOptions {
-    // Give the benchmark code access to internal definitions.
-    val compileKotlin: KotlinCompile by tasks
-    freeCompilerArgs += "-Xfriend-paths=${compileKotlin.destinationDir}"
+kotlin {
+  jvm {
+    compilations {
+      val main by getting
+      val workflowNode by creating {
+        kotlinOptions {
+          val compileKotlinJvm: KotlinCompile by tasks
+          freeCompilerArgs += "-Xfriend-paths=${compileKotlinJvm.destinationDir}"
+        }
+        defaultSourceSet {
+          dependencies {
+            implementation(main.compileDependencyFiles + main.output.classesDirs)
+            implementation("org.jetbrains.kotlinx:kotlinx-benchmark-runtime:0.4.2")
+          }
+        }
+      }
+    }
+    withJava()
   }
-}
-
-dependencies {
-  compileOnly(libs.jetbrains.annotations)
 
-  api(project(":workflow-core"))
-  api(libs.kotlin.jdk6)
-  api(libs.kotlinx.coroutines.core)
+  sourceSets {
+    val jvmMain by getting {
+      dependencies {
+        compileOnly(libs.jetbrains.annotations)
 
-  testImplementation(libs.kotlinx.coroutines.test)
-  testImplementation(libs.kotlin.test.jdk)
-  testImplementation(libs.kotlin.reflect)
+        api(project(":workflow-core"))
+        api(libs.kotlin.jdk6)
+        api(libs.kotlinx.coroutines.core)
+      }
+    }
+    val jvmTest by getting {
+      dependencies {
+        implementation(libs.kotlinx.coroutines.test)
+        implementation(libs.kotlin.test.jdk)
+        implementation(libs.kotlin.reflect)
+      }
+    }
+  }
+}
 
-  // These dependencies will be available on the classpath for source inside src/jmh.
-  "jmh"(libs.kotlin.jdk6)
-  "jmh"(libs.jmh.core)
-  "jmh"(libs.jmh.generator)
+benchmark {
+  targets {
+    register("jvmWorkflowNode") {
+      this as JvmBenchmarkTarget
+      jmhVersion = "1.35"
```

Comment:
Yep.  Revert the line you removed from the "plugins" section, then you can reference here it like:

```kotlin
jmhVersion = libs.versions.jmh.get()
```

Note that since the catalog is generated, you have to do a gradle sync before the IDE recognizes the new definition.

Reply:
Thanks